solutions/cses-1163.mdx

```diff
@@ -105,3 +107,111 @@ int main(){
 	return 0;
 }
 ```
+
+</CPPSection>
+<JavaSection>
+## Implementation
+[Unofficial Editorial (C++)](https://codeforces.com/blog/entry/83295)
+
+**Time Complexity:** $\mathcal{O}(n)$
+
+For each traffic light that is added, we want to find the largest current gap
+between traffic lights.
+
+First, let's get our values. We'll use a set to represent the street itself, its
+starting and ending points, and every traffic light along it. We'll also use an
+integer array to store and read the positions of the traffic lights, since we
+can't read values from sets in Java. We'll store our outputs in another array
+too.
+
+Now, to actually solve the problem! We'll start by trying to find the maximum
+gap once all the traffic lights are added. This is the last thing we'll output,
+so we'll add it to the end of our output array. Then, we'll remove traffic
+lights in the reverse order to how they were added, and find the gap each
+removal creates.
+
+This gap is just the distance between the two street coordinates (either a
+traffic light or the beginning or ending of the street) next to a given traffic
+light stored in our set, so we can use our set to find these values and subtract
+them to get our gap.
+
+But this gap may not be the maximum gap. We'll compare this to the gap we found
+once all the traffic lights are added, and set the max gap to the greater value.
+We'll then set the next lowest element in the output array to this value, which
+will represent the greatest gap prior to adding the traffic light we just
+removed.
+
+Once we've gone through and removed every traffic light and added every max gap
+to our output array, we'll iterate through the output array and print out each
+element. Boom-we're done!
```

Comment:
```suggestion
```
i love your tone here, but unfortunately it's inconsistent with the usual tone- the usual tone should be ~~condescending~~ formal (also this part isn't really necessary)